raiden/storage/sqlite.py

```diff
@@ -136,6 +143,52 @@ def get_latest_state_snapshot(self) -> Optional[Tuple[int, Any]]:
 
         return result
 
+    def get_latest_event_by_data_field(self, field: str, value: str):
+        """ Return all state changes filtered by a named field and value."""
+        cursor = self.conn.cursor()
+
+        cursor.execute(
+            "SELECT data FROM state_events WHERE "
+            f"json_extract(data, '$.{field}')=? "
+            "ORDER BY identifier DESC LIMIT 1",
+            (value,),
+        )
+
+        result = None
+        try:
+            row = cursor.fetchone()
+            if row:
+                result = self.serializer.deserialize(row[0])
+        except AttributeError:
+            raise InvalidDBData(
```

Comment:
Please make this an unrecoverable error (it is not ATM)

Reply:
How is it not an unrecoverable error? It will crash the node.

Reply:
we have a class named `RaidenUnrecoverableError`:

https://github.com/raiden-network/raiden/blob/39a2f547b7dfc9b201dfca6b12ebd80d9dc5d932/raiden/exceptions.py#L12

and `InvalidDBData` doesn't inherit from it

https://github.com/raiden-network/raiden/blob/39a2f547b7dfc9b201dfca6b12ebd80d9dc5d932/raiden/exceptions.py#L173

Reply:
Ah I see. What I meant is that since it will crash it's an unrecoverable error.

Sure we can do that. @rakanalh please make it inherit from `RaidenUnrecoverableError`.